ecs-cli/integ/e2e/ec2_task_test.go

```diff
@@ -0,0 +1,120 @@
+// +build integ
+
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package e2e
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/aws/amazon-ecs-cli/ecs-cli/integ/ecs"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/aws/amazon-ecs-cli/ecs-cli/integ/cmd"
+)
+
+// TestCreateClusterWithEC2Task runs the sequence of ecs-cli commands from
+// the EC2 tutorial: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-cli-tutorial-ec2.html
+func TestCreateClusterWithEC2Task(t *testing.T) {
+	t.Parallel()
+
+	// Create the cluster
+	conf := cmd.TestEC2TutorialConfig(t)
+	cmd.TestUp(t,
+		conf,
+		cmd.WithCapabilityIAM(),
+		cmd.WithInstanceType("t2.medium"),
+		cmd.WithSize(2),
+	)
+	ecs.TestClusterSize(t, conf.ClusterName, 2)
+
+	// Create the files for a task definition
+	project := cmd.NewProject("e2e-ec2-tutorial-taskdef", conf.ConfigName)
+	project.ComposeFileName = createEC2TutorialComposeFile(t)
+	project.ECSParamsFileName = createEC2TutorialECSParamsFile(t)
+	defer os.Remove(project.ComposeFileName)
+	defer os.Remove(project.ECSParamsFileName)
+
+	// Create a new task
+	cmd.TestTaskUp(t, project)
+	ecs.TestListTasks(t, conf.ClusterName, 1)
+	cmd.TestPsRunning(t, project, 2)
+
+	// Increase the number of running tasks
+	cmd.TestTaskScale(t, project, 2)
+	ecs.TestListTasks(t, conf.ClusterName, 2)
+	cmd.TestPsRunning(t, project, 4)
```

Comment:
The numbers here confuse me... after TestTaskUp, why are 2 containers expected in PS?

And after scaling to 2 tasks, why are 4 containers expected in PS?

Reply:
Yeah, I can totally understand the confusion. 
If you look at the compose file, we have two containers listed out. So that 1 task with two containers. When we scale the number of tasks to 2, then we end up with 4 running containers in the cluster.

I'll add comments explaining this in the integ test.